src/rules/deprecation.ts

```diff
@@ -47,34 +78,9 @@ export default createRule<Options, MessageIds>({
   },
   defaultOptions: [],
   create(context) {
-    const services = ESLintUtils.getParserServices(context);
-
     return {
-      Identifier: (id: TSESTree.Identifier) => {
-        const insideImportExport = context
-          .getAncestors()
-          .some(anc => anc.type.includes('Import'));
-        if (
-          insideImportExport ||
-          (id.type === 'Identifier' && isDeclaration(id, context))
-        ) {
-          // Don't consider deprecations if inside import/export.
-          // Also, the spot where something is declared is never deprecated.
-          return;
-        }
-
-        const deprecation = getDeprecation(id, services, context);
-        if (deprecation) {
-          context.report({
-            node: id,
-            messageId: 'deprecated',
-            data: {
-              name: id.name,
-              reason: deprecation.reason,
-            },
-          });
-        }
-      },
+      Identifier: ruleCreator(context),
+      JSXIdentifier: ruleCreator(context),
```

Comment:
Please call creation function once and reuse the rule for both props using constant.